### PR TITLE
Document changes in reconcile commands

### PR DESCRIPTION
**Pull request by simo5**
Fixes https://github.com/openshift/origin/issues/15412

### CODE REVIEW

**Comment by simo5** [comment]
@enj PTAL

File: architecture/additional_concepts/authorization.adoc [hunk 1]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
 === Updating Cluster Roles
 
 After any xref:../../install_config/upgrading/index.adoc#install-config-upgrading-index[{product-title} cluster
-upgrade], the recommended default roles may have been updated. See
+upgrade], the default roles are updated and automatically reconciled when the
+server is started. Additionally see
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: Additionally, see

File: architecture/additional_concepts/authorization.adoc [hunk 2]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
...
 xref:../../install_config/upgrading/manual_upgrades.adoc#updating-policy-definitions[Updating
-Policy Definitions] for instructions on getting to the new recommendations
+Policy Definitions] for instructions on getting other recommendations
 using:
 
 ----
 $ oc adm policy reconcile-cluster-roles
 ----
+
+=== Applying custom Roles and Permissions
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: Applying Custom

File: architecture/additional_concepts/authorization.adoc [hunk 3]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
...
+server is started. Additionally see
 xref:../../install_config/upgrading/manual_upgrades.adoc#updating-policy-definitions[Updating
-Policy Definitions] for instructions on getting to the new recommendations
+Policy Definitions] for instructions on getting other recommendations
 using:
 
 ----
 $ oc adm policy reconcile-cluster-roles
 ----
+
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: Need to add a section ID before the section heading:

[[applying-custom-roles-and-permissions]]

File: architecture/additional_concepts/authorization.adoc [hunk 4]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
...
+=== Applying custom Roles and Permissions
+
+To add or update custom roles and permissions it is strongly recommended to use
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: permissions, it is strongly recommended that you use

File: architecture/additional_concepts/authorization.adoc [hunk 5]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
...
+the following command:
+
+----
+# oc auth reconcile -f FILE
+----
+
+This command insures that new permission are applied "properly" in a way that
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: s/insures/ensures
s/permission/permissions
s/"properly"/properly

File: architecture/additional_concepts/authorization.adoc [hunk 6]
```diff
@@ -273,14 +273,30 @@ ifdef::openshift-enterprise,openshift-origin[]
...
+will not break other clients. This is done internally by computing logical covers
+operations between rule sets which is something that cannot be simply done via
+a json merge on policy files.
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authorization.adoc]: sets, which you cannot do via a JSON merge on policy files.

File: install_config/upgrading/manual_upgrades.adoc [hunk 7]
```diff
@@ -423,10 +423,10 @@ updated.
 [[updating-policy-definitions]]
 == Updating Policy Definitions
 
-After a cluster upgrade, the recommended
+After a cluster upgrade, the default roles
 xref:../../architecture/additional_concepts/authorization.adoc#roles[default
-cluster roles] may be updated. To check if an update is recommended for
-your environment, you can run:
+cluster roles] are automatically updated. To check if all defaults are set as
+recommended for your environment, you can run:
```
Comment (ahardin-rh) [inline install_config/upgrading/manual_upgrades.adoc]: s/you can run/run

File: install_config/upgrading/manual_upgrades.adoc [hunk 8]
```diff
@@ -436,8 +436,10 @@ your environment, you can run:
 ====
 If you have customized default cluster roles and want to ensure a role reconciliation
 does not modify those customized roles, annotate them with `openshift.io/reconcile-protect`
-set to `true`. Doing so means you are responsible for manually updating those roles with
-any new or required permissions during upgrades.
+set to `true` when using the old Openshift policy format. When using the new RBAC
+roles use `rbac.authorization.kubernetes.io/autoupdate` set to `false` instead.
```
Comment (ahardin-rh) [inline install_config/upgrading/manual_upgrades.adoc]: roles, use

and set to

File: install_config/upgrading/manual_upgrades.adoc [hunk 9]
```diff
@@ -436,8 +436,10 @@ your environment, you can run:
...
+Doing so means you are responsible for manually updating those roles with any new or
```
Comment (ahardin-rh) [inline install_config/upgrading/manual_upgrades.adoc]: In doing so, you are responsible

**Comment by simo5** [comment]
Should have fixed all nits.
@ahardin-rh PTAL

**Comment by ahardin-rh** [comment]
@simo5 LGTM. Per our publishing process, can you please squash the commits into a single commit? Also, looks like this is targeting 3.7, correct? Thanks again!

**Comment by simo5** [comment]
@ahardin-rh squashed, yes 3.7

**Comment by ahardin-rh** [comment]
No revision history needed

**Comment by vikram-redhat** [comment]
Hey @ahardin-rh - does this need to be cherry picked to dedicated and online branches?

**Comment by enj** [comment]
Just to be clear, these docs only apply in situations where a single entity controls a cluster (another way to think about it is that it only applies to administrators).  Thus users of the online multi-tenant environments have no need for this information (because we handle those upgrades and manage the authz data).  It only applies to users who manage their own cluster (and its upgrades).

**Comment by ahardin-rh** [comment]
@vikram-redhat Those cherry-picks happened in https://github.com/openshift/openshift-docs/pull/5825 and https://github.com/openshift/openshift-docs/pull/5823. Not sure why the xref was wonky for those. 

@enj Thanks! I have ifdef statements in place, which will keep this information from appearing in Online.

**Pull request merged**